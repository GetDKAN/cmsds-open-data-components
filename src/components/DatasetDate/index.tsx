import React from 'react';
import DatasetDateItem from '../DatasetDateItem';

type DateObject = {
  modified?: string;
  released?: string;
  refresh?: string;
};

type DatasetDateProps = {
  /**
   * Released and modified date strings
   */
  date: DateObject;
  /**
   * Apply bold style to updated label
   */
  updatedBoldLabel?: boolean;
  /**
   * Apply bold style to released label
   */
  releasedBoldLabel?: boolean;
  /**
   * Apply bold style to refresh label
   */
  refreshBoldLabel?: boolean;
  /**
   * Display tooltips or not
   */
  displayTooltips?: boolean;
};

const DatasetDate: React.FC<DatasetDateProps> = (props) => {
  const {
    date,
    updatedBoldLabel = false,
    releasedBoldLabel = false,
    refreshBoldLabel = false,
    displayTooltips = true
  } = props;

  const { modified, released, refresh } = date;

  let count = 0;
  if (released) count++;
  if (modified) count++;
  if (refresh) count++;

  return (
    <div className='dataset-date'>
      {modified && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='modified' 
          date={modified} 
          boldLabel={updatedBoldLabel} 
        />
      )}
      {count > 1 && <span className='bullet-point'>&bull;</span>}
      {released && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='released' 
          date={released} 
          boldLabel={releasedBoldLabel} 
        />
      )}
      {count > 2 && <span className='bullet-point'>&bull;</span>}
      {refresh && (
        <DatasetDateItem 
          displayTooltips={displayTooltips} 
          type='refresh' 
          date={refresh} 
          boldLabel={refreshBoldLabel} 
        />
      )}
    </div>
  );
};

export default DatasetDate; 