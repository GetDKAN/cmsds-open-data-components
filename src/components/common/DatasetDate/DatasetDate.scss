@use "../../../scss/modules/modules";
@use "../../../scss/modules/colormap";

.dataset-date {
  display: flex;

  .dataset-date-item {
    color: colormap.$charcoal-90;
    font-size: 1.4rem;
    white-space: nowrap;

    &.bold-label > span {
      font-weight: 700;
    }
    .ds-c-tooltip__container {
      position: relative;
      bottom: 10px;
      width: 20px;
      display: inline-block;
    }
    .ds-c-tooltip__trigger-icon {
      padding: 0
    }
    .ds-c-tooltip {
      max-width: revert !important;
      font-size: var(--theme-font-size-md, 16px);
      font-style: normal;
      font-weight: 700;
      line-height: 150%; /* 24px */
    }
  }
}

@media screen and (max-width: 600px) {
  .dataset-date {
    flex-direction: column;

    .dataset-date-item {
      margin-bottom: 0.5rem;
    }

    .bullet-point {
      display: none;
    }
  }
}
