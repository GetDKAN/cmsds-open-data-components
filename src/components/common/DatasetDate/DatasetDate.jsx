import React from 'react'
import PropTypes from 'prop-types'
import DatasetDateItem from './DatasetDateItem/DatasetDateItem'

import './DatasetDate.scss'

const DatasetDate = (props) => {
  const {
    date,
    updatedBoldLabel = false,
    releasedBoldLabel = false,
    refreshBoldLabel = false,
    displayTooltips = true
  } = props

  const { modified, released, refresh } = date

  let count = 0
  released && count++
  modified && count++
  refresh && count++

  return (
    <div className='dataset-date'>
      {modified && <DatasetDateItem displayTooltips={displayTooltips} type='modified' date={modified} boldLabel={updatedBoldLabel} />}
      {count > 1 && <span className='bullet-point'>&bull;</span>}
      {released && <DatasetDateItem displayTooltips={displayTooltips} type='released' date={released} boldLabel={releasedBoldLabel} />}
      {count > 2 && <span className='bullet-point'>&bull;</span>}
      {refresh && <DatasetDateItem displayTooltips={displayTooltips} type='refresh' date={refresh} boldLabel={refreshBoldLabel} />}
    </div>
  )
}

DatasetDate.propTypes = {
  /**
   * Released and modified date strings
   */
  date: PropTypes.shape({
    modified: PropTypes.string,
    released: PropTypes.string,
    refresh: PropTypes.string
  }),
  /**
   * Apply bold style to updated label
   */
  updatedBoldLabel: PropTypes.bool,
  /**
   * Apply bold style to released label
   */
  releasedBoldLabel: PropTypes.bool,
  /**
   * Apply bold style to refresh label
   */
  refreshBoldLabel: PropTypes.bool,
  /**
   * Display tooltips or not
   */
  displayTooltips: PropTypes.bool
}
DatasetDate.displayName = 'DatasetDate'
export default DatasetDate
