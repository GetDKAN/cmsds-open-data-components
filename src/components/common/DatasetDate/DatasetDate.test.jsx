import renderer from 'react-test-renderer'
import DatasetDate from './DatasetDate'

describe('DatasetDate component.', () => {
  it('Matches snapshot.', () => {
    const renderedDatasetDate = renderer.create(
      <DatasetDate date={{
        modified: '2023-02-01',
        released: '2023-01-01'
      }} />
    ).toJSON()

    expect(renderedDatasetDate).toMatchSnapshot()
  })
})
