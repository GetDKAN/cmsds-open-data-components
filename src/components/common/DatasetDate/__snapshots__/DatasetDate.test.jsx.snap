// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatasetDate component. > Matches snapshot. 1`] = `
<div
  className="dataset-date"
>
  <div
    className="dataset-date-item"
  >
    <span
      className="dataset-data-item-label"
    >
      <span>
        Last Modified
      </span>
      : 
      February 1, 2023
    </span>
    <div
      className="ds-c-tooltip__container"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-describedby="tooltip-trigger--1"
        aria-label="Last Modified"
        className="ds-c-tooltip__trigger ds-c-tooltip__trigger-icon ds-u-display--inline ds-u-padding-left--0 ds-uw-padding-right--0"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          className="ds-c-tooltip-icon__container"
        >
          <svg
            aria-hidden={true}
            className="ds-c-icon ds-c-icon--info-circle-thin ds-c-tooltip-icon"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 16c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8zm1-3.7V6.4H7v5.9h2zM7 4.9c0 .6.3.9 1 .9s1-.3 1-.9c0-.3-.1-.5-.2-.7-.2-.1-.5-.2-.8-.2-.3 0-.6.1-.8.2-.1.2-.2.4-.2.7z"
              fillRule="nonzero"
            />
          </svg>
        </span>
      </button>
      <div
        aria-hidden={true}
        className="ds-c-tooltip"
        data-placement="top"
        id="tooltip-trigger--1"
        onBlur={[Function]}
        role="tooltip"
        style={
          {
            "maxWidth": "300px",
            "zIndex": 9999,
          }
        }
        tabIndex={null}
      >
        <span
          className="ds-c-tooltip__arrow"
          data-popper-arrow={true}
        />
        <div
          className="ds-c-tooltip__content"
        >
          <span
            dangerouslySetInnerHTML={
              {
                "__html": "<strong>Last Modified: </strong> The date the<br/>dataset was last updated.",
              }
            }
          />
        </div>
        <span
          className="ds-c-tooltip__interactive-border"
          style={
            {
              "border": "15px solid transparent",
              "left": "-15px",
              "top": "-15px",
              "zIndex": -999,
            }
          }
        />
      </div>
    </div>
  </div>
  <span
    className="bullet-point"
  >
    â€¢
  </span>
  <div
    className="dataset-date-item"
  >
    <span
      className="dataset-data-item-label"
    >
      <span>
        Released
      </span>
      : 
      January 1, 2023
    </span>
    <div
      className="ds-c-tooltip__container"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-describedby="tooltip-trigger--3"
        aria-label="Released"
        className="ds-c-tooltip__trigger ds-c-tooltip__trigger-icon ds-u-display--inline ds-u-padding-left--0 ds-uw-padding-right--0"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          className="ds-c-tooltip-icon__container"
        >
          <svg
            aria-hidden={true}
            className="ds-c-icon ds-c-icon--info-circle-thin ds-c-tooltip-icon"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 16c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8zm1-3.7V6.4H7v5.9h2zM7 4.9c0 .6.3.9 1 .9s1-.3 1-.9c0-.3-.1-.5-.2-.7-.2-.1-.5-.2-.8-.2-.3 0-.6.1-.8.2-.1.2-.2.4-.2.7z"
              fillRule="nonzero"
            />
          </svg>
        </span>
      </button>
      <div
        aria-hidden={true}
        className="ds-c-tooltip"
        data-placement="top"
        id="tooltip-trigger--3"
        onBlur={[Function]}
        role="tooltip"
        style={
          {
            "maxWidth": "300px",
            "zIndex": 9999,
          }
        }
        tabIndex={null}
      >
        <span
          className="ds-c-tooltip__arrow"
          data-popper-arrow={true}
        />
        <div
          className="ds-c-tooltip__content"
        >
          <span
            dangerouslySetInnerHTML={
              {
                "__html": "<strong>Released: </strong> The date the most<br/>recent dataset was made available<br/>to the public.",
              }
            }
          />
        </div>
        <span
          className="ds-c-tooltip__interactive-border"
          style={
            {
              "border": "15px solid transparent",
              "left": "-15px",
              "top": "-15px",
              "zIndex": -999,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
